sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/kafka/AzureKafkaOAuth2BinderConfigurationTests.java

```diff
@@ -51,70 +76,112 @@ void shouldNotConfigureBPPWithoutKafkaMessageChannelBinder() {
                 .withPropertyValues(
                         SPRING_CLOUD_STREAM_KAFKA_PROPERTIES_PREFIX + CLIENT_ID + "=cloud-client-id",
                         SPRING_CLOUD_STREAM_KAFKA_CONSUMER_PROPERTIES_PREFIX + CLIENT_ID + "=cloud-consumer-client-id",
                         SPRING_BOOT_KAFKA_PROPERTIES_PREFIX + CLIENT_ID + "=kafka-client-id",
                         SPRING_BOOT_KAFKA_PRODUCER_PROPERTIES_PREFIX + CLIENT_ID + "=kafka-producer-client-id",
-                        "spring.cloud.azure.credential.client-id=azure-client-id"
+                        "spring.cloud.azure.credential.client-id=azure-client-id",
+                        "spring.kafka.bootstrap-servers=myehnamespace.servicebus.windows.net:9093"
                 )
                 .run(context -> {
-                    assertThat(context).hasSingleBean(AzureKafkaSpringCloudStreamConfiguration.class);
-                    assertThat(context).hasSingleBean(KafkaBinderConfigurationPropertiesBeanPostProcessor.class);
-                    assertThat(context).hasSingleBean(KafkaBinderConfigurationProperties.class);
-                    assertThat(context).hasSingleBean(KafkaTopicProvisioner.class);
-
-                    KafkaBinderConfigurationProperties kafkaProperties = context.getBean(KafkaBinderConfigurationProperties.class);
-
-                    Map<String, String> kafkaConfiguration = kafkaProperties.getConfiguration();
-                    assertEquals("cloud-client-id", kafkaConfiguration.get(CLIENT_ID));
-
-                    KafkaTopicProvisioner kafkaTopicProvisioner = context.getBean(KafkaTopicProvisioner.class);
-                    Map<String, Object> adminClientProperties = (Map<String, Object>) ReflectionTestUtils.getField(kafkaTopicProvisioner,
-                            "adminClientProperties");
-                    shouldConfigureOAuthProperties(adminClientProperties);
-
-                    Map<String, Object> consumerConfiguration = kafkaProperties.mergedConsumerConfiguration();
-                    assertEquals("cloud-consumer-client-id", consumerConfiguration.get(CLIENT_ID));
-                    shouldConfigureOAuthProperties(consumerConfiguration);
-
-                    Map<String, Object> producerConfiguration = kafkaProperties.mergedProducerConfiguration();
-                    assertEquals("kafka-producer-client-id", producerConfiguration.get(CLIENT_ID));
-                    shouldConfigureOAuthProperties(producerConfiguration);
+                    KafkaBinderConfigurationProperties kafkaSpringProperties = getKafkaSpringProperties(context);
+                    assertConsumerPropsConfigured(kafkaSpringProperties, CLIENT_ID, CLIENT_ID + "=\"cloud-consumer-client-id\"");
+                    assertProducerPropsConfigured(kafkaSpringProperties, CLIENT_ID, CLIENT_ID + "=\"cloud-client-id\"");
+                    assertAdminPropsConfigured(kafkaSpringProperties, CLIENT_ID, CLIENT_ID + "=\"cloud-client-id\"");
```

Comment:
can we remove these backslashes?

Reply:
ok, i got it, it's Kafka jaas format?

Reply:
but can we improve the code? Can't we add this logic in these methods assertAdminPropsConfigured?

Reply:
To combine this comment and the above one, how about we changing these APIs as `assertAdminPropsConfigured(kafkaSpringProperties, CLIENT_ID, "cloud-client-id")`

Reply:
I once saw Kafka's code that it adds the quotations for all values from Jaas options, let me try to find that code